modules/environments/azure_devops.tf

```diff
@@ -0,0 +1,12 @@
+resource "azuredevops_serviceendpoint_azurerm" "endpoint" {
+  count                 = var.pipeline_environment == "prod" ? 1 : 0
+  project_id            = var.project_id
+  service_endpoint_name = "OPS-APPROVAL-GATE-${upper(var.env)}-ENVS"
+  credentials {
+    serviceprincipalid  = azuread_service_principal.sp.application_id
+    serviceprincipalkey = azuread_application_password.token.value
+  }
+  azurerm_spn_tenantid      = data.azurerm_client_config.current.tenant_id
+  azurerm_subscription_id   = var.env == "ptl" ? data.azurerm_subscriptions.service_endpoint_subscription[0].subscriptions[1].subscription_id : data.azurerm_subscriptions.service_endpoint_subscription[0].subscriptions[0].subscription_id
```

Comment:
add comment explaining why this is done as it is?

I believe it's because the subscriptions datasource only has prefix support and with this environment there's multiple matches? 
can you create a feature request upstream as well which could just be linked to?